### DISCUSSION TITLE
oh-my-posh causes my Powershell to hang

### DISCUSSION THREAD

**Discussion by fgheysels** [JanDeDobbeleer/oh-my-posh#2760] [General]
I'm setting up a new laptop that is running Windows 11.
I have Powershell Core 7.2.6 installed, and I have installed:
- PSReadLine 2.2.6
- Oh-my-Posh via the guide that is found [here](https://ohmyposh.dev/docs/installation/windows)

I have this line in my $PROFILE

```
oh-my-posh --init --shell pwsh --config ~/frederik.omp.json | Invoke-Expression
```
When I start a new Terminal or Powershell Core session, I can see that my profile is loaded, but then it just hangs.  No prompt is displayed and I cannot type anything

![image](https://user-images.githubusercontent.com/3605786/189157405-04a89d2a-a0b3-4f1a-8501-c668d41a36ed.png)

Note that the message which displays how long it took to load the profile, is not always shown:

![image](https://user-images.githubusercontent.com/3605786/189154181-0b07bdcf-dc77-47a8-b77f-5d8a46c0d4fc.png)

I have configured an exclusion in Windows Security for the `...\AppData\Local\Programs\oh-my-posh\bin\oh-my-posh.exe` file, but this doesn't seem to help.
I've then also tried what is mentionned [here](https://ohmyposh.dev/docs/installation/prompt) regarding the replacement of the Invoke-Expression statement with a ScriptBlock, but this also doesn't work. 

I can't find anything in Windows EventLogs that could point my in a certain direction.
Strange thing is that sometimes I get the 'oh my posh' prompt, and I can execute a command, but after one or two commands, Powershell hangs.  However, most of the times (9 out of 10) PS hangs immediately on startup.

**Comment by lewis-yeung** [comment]
@fgheysels Comment out the Oh My Posh initialization line in your PowerShell profile, run `oh-my-posh debug -c ~/frederik.omp.json` and see if it also hangs.

**Comment by fgheysels** [comment]
I've done that, it doesn't hang and I also see no errors or warnings.

**Comment by fgheysels** [comment]
But, if I just wait long enough, things start to move ...

![image](https://user-images.githubusercontent.com/3605786/189169328-5fa95ffa-ce78-4e7f-b2f7-638db31c5d96.png)

Unfortunately, that is not doable.  17 minutes to wait before the prompt is usable is just a bit too long :D

In another situation, I have this:
- the prompt shows quickly, but after entering a command, it just hangs:
![image](https://user-images.githubusercontent.com/3605786/189186225-73002651-ac3e-4c4c-927b-62b556feea06.png)

**Reply by JanDeDobbeleer** [reply]
When you do get a prompt, we need the output of `oh-my-posh debug`. That should show where the time is spent. I would also try a profile that only has the oh-my-posh init line (and nothing else).

**Comment by fgheysels** [comment]
I'll change my profile; it currently has only 2 lines (oh-my-posh and terminal-icons).
I've also seen that Bitlocker is enabled via a company policy on the laptop.  Could that be an issue ?

**Reply by JanDeDobbeleer** [reply]
Shouldn't be. Unless the go executable can't access the folders or something else.

**Comment by fgheysels** [comment]
It is hanging / spending a lot of time at this point:

![image](https://user-images.githubusercontent.com/3605786/189195984-e8547a12-f3bf-4efb-8e30-a81b687a5947.png)

I can also see that the segments in my prompt have changed once the command finished:

![image](https://user-images.githubusercontent.com/3605786/189196384-42e0d98e-1144-4cdb-8066-9311a30c15f0.png)


![image](https://user-images.githubusercontent.com/3605786/189196586-968bddc4-d127-424a-9d32-5b424e971a28.png)

**Reply by JanDeDobbeleer** [reply]
@fgheysels this is device related. Not sure how/why, but your device blocks the execution even with it having set the exclusion. Oh-my-posh runs 12ms start to finish so Windows isn't starting it immediately.

**Reply by fgheysels** [reply]
@JanDeDobbeleer And what about the fact that the prompt changes after execution of that debug statement ?

**Reply by JanDeDobbeleer** [reply]
@fgheysels debug doesn't resolve the config flag in PowerShell like we normally do. It's also not necessary.

**Reply by lewis-yeung** [reply]
> And what about the fact that the prompt changes after execution of that debug statement ?

@fgheysels That means OMP failed to download a default remote config so it used an internal fallback "no config".

**Comment by fgheysels** [comment]
I suspect I found what was happening.  The path to my omp.json file was not correct so maybe it sometimes took that much time to find the default / fallback config file ?

Anyway, I haven't experienced any delays anymore since I corrected the path.

**Reply by lewis-yeung** [reply]
@fgheysels When the specified config path is invalid, OMP will try to download a remote config (default.omp.json), so an HTTP request is created for this. I think it probably spent quite a long while on your end, though 17+ minutes is unimaginable. 😅

**Comment by gafda** [comment]
Hey everybody!

I'm facing this same problem in WSL2 instance of Ubuntu 22.04.

After installation, I downloaded a theme right from the official GIT repo. (_**Length:** 88778 bytes_)
When I try to use it, it just hags. I had to kill the WSL instance to get access to the shell command line as you can see in the screenshot.
After that, I went to check on the theme itself, and as you can see... something had gone **Really** wrong...

![image](https://github.com/JanDeDobbeleer/oh-my-posh/assets/4069765/b9b63e3d-5f44-4e44-8f41-293dfc7132da)

Does anyone have a solution for this?

Oh-my-posh >> v19.11.1

Later note: Updating to v19.11.7 didn't fix it. It has the same symptom.

**Reply by JanDeDobbeleer** [reply]
You downloaded the GitHub web page of the theme, not the actual file. You need to look at the raw file and download that.

**Reply by gafda** [reply]
Omg! This is embarrassing... You're absolutely right! tysm!

Still... Why didn't it just fail fast saying the Theme was invalid or something like that? I mean... It Never recovered... :(
